Fixed: Download notification disappears when pausing a download.

MohitMaliFtechiz opened the pull request:
Fixes #4207 

* The Fetch library automatically removes the ongoing download notification when a download is paused. To improve the user experience, we have implemented custom logic to display a notification for paused downloads, allowing users to easily resume them.
* Additionally, Android 14 introduces a change that prevents non-dismissible notifications see for more details https://developer.android.com/about/versions/14/behavior-changes-all#non-dismissable-notifications. This means users can now dismiss notifications by swiping left or right. Since our app uses a foreground service for downloads, we must always display a notification. To handle this, we have implemented a mechanism that automatically adds the last notification to our foreground service to satisfy its need when no download is ongoing, and the user can still cancel it on Android 14 and above.

https://github.com/user-attachments/assets/4db20155-32fd-418a-b121-25ba51b6a4a8

kelson42 commented:
I don't rhink this is a good idea to bring back a notification the user has explicitly dismiss.

MohitMaliFtechiz commented:
> I don't rhink this is a good idea to bring back a notification the user has explicitly dismiss. We should not bring it back and user should rhen go to kiwix to handle his download.

@kelson42 I have removed the again showing of notification and updated the PR description.
